tests/test_camera_functions.py

```diff
@@ -167,12 +168,32 @@ def test_recent_video_clips(self, mock_resp):
         self.camera.sync.last_records["foobar"].append(record2)
         record1 = {"clip": "/clip1", "time": "2022-12-01 00:00:00+00:00"}
         self.camera.sync.last_records["foobar"].append(record1)
-        self.camera.update_images(config)
+        self.camera.sync.motion["foobar"] = True
+        self.camera.update_images(config, expire_clips=False)
```

Comment:
In short, I want to get _all_ the clips the cameras produce.  In my experience, I was missing many clips created by a camera - like 1 in 3 or 1 in 4.  This is when I had the Blink cloud storage trial active.  Only getting a third of the clips was frustrating me, especially when the one that got downloaded didn't have the subject that triggered the motion in it (by chance of course).  Does this not happen for you?

There are latencies / race conditions with detecting motion, updating the urls, and downloading the clips.  I am attempting to mitigate the loss of video clips due to those latencies and race conditions.  (There appear to still be race conditions, but this is a step in the right direction).

In my opinion, the motion detection is of limited value.  The greater value (IMO) is to have all the clips downloaded onto my local USB drive and easily reviewed using the [gallery card](https://github.com/TarheelGrad1998/gallery-card).  I want to not need to open the Blink app (it is pretty terrible, especially for local storage viewing).  Using this approach makes HA a reasonable replacement for the Blink app.  And I think that HA Blink users will like it.

Since this is used as a cloud-polling integration, I plan on using the polling mechanism:

1. Call `blink.refresh()` on a schedule to detect if there are any new videos available
2. Have blinkpy sort those clips by camera in each camera's `recent_clips`
3. Call `camera.save_recent_clips()` on a schedule to download the clips
4. Optionally send myself a push notification if there are new clips to review.  The Blink app will send me a notification too anyway - so this may be redundant.

The `expire_clips` is a way to keep the list from growing too large if the user decides not to download the clips for what ever reason.  The expiry is based on real time, and I added the `expire_clips=False` so that existing tests will still pass.

Have you been able to test this branch using your cloud storage mechanism?

Reply:
Have you been able to test this branch using your cloud storage mechanism?

Reply:
Also, I have been running this setup in my DEV HA instance for a couple of days now with pretty good results.
```
-rw-rw-r-- 1 brent.perdue brent.perdue 2158285 Dec  6 11:55 20221206_115418_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  786634 Dec  6 12:01 20221206_120004_SouthYard.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2999327 Dec  6 12:01 20221206_120009_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  732801 Dec  6 12:10 20221206_120903_SouthYard.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3207364 Dec  6 12:11 20221206_120911_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  442649 Dec  6 12:10 20221206_120922_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  416541 Dec  6 12:13 20221206_121211_LivingRoom.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  413550 Dec  6 12:13 20221206_121214_SouthYard.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3676150 Dec  6 12:13 20221206_121215_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  500334 Dec  6 12:13 20221206_121232_LivingRoom.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  399890 Dec  6 12:13 20221206_121240_SouthYard.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  448378 Dec  6 12:13 20221206_121259_LivingRoom.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3230251 Dec  6 13:45 20221206_134304_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  780804 Dec  6 13:44 20221206_134322_SouthYard.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2990693 Dec  6 13:46 20221206_134430_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3204528 Dec  6 13:48 20221206_134609_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2928218 Dec  6 13:50 20221206_134755_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3236013 Dec  6 13:51 20221206_134835_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2989071 Dec  6 13:50 20221206_134917_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2952317 Dec  6 13:52 20221206_135034_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2961291 Dec  6 13:54 20221206_135155_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3194732 Dec  6 13:55 20221206_135234_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2999855 Dec  6 13:54 20221206_135315_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2753500 Dec  6 13:57 20221206_135354_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3296268 Dec  6 13:55 20221206_135439_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2900008 Dec  6 13:58 20221206_135556_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2987131 Dec  6 13:59 20221206_135730_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3217854 Dec  6 14:00 20221206_135810_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3222289 Dec  6 14:01 20221206_135930_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2917302 Dec  6 14:03 20221206_140046_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3156686 Dec  6 14:36 20221206_143353_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3220304 Dec  6 14:35 20221206_143446_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2987934 Dec  6 14:41 20221206_143841_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3234020 Dec  6 14:40 20221206_143937_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3230934 Dec  6 14:46 20221206_144408_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3182058 Dec  6 14:50 20221206_144718_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3000046 Dec  6 14:48 20221206_144754_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  372895 Dec  6 15:21 20221206_151956_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  477063 Dec  6 15:28 20221206_152659_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  342035 Dec  6 15:31 20221206_152937_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3000655 Dec  6 15:38 20221206_153605_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3120512 Dec  6 15:39 20221206_153645_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3196797 Dec  6 15:38 20221206_153723_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3178603 Dec  6 15:40 20221206_153805_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3186979 Dec  6 15:43 20221206_154053_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3232248 Dec  6 15:47 20221206_154554_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2943901 Dec  6 15:49 20221206_154632_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2969149 Dec  6 15:49 20221206_154801_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  443506 Dec  7 09:18 20221207_091625_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3231245 Dec  7 09:21 20221207_091851_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3159457 Dec  7 09:24 20221207_091927_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3003808 Dec  7 09:27 20221207_092430_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3206215 Dec  7 09:30 20221207_092751_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3005781 Dec  7 09:30 20221207_092840_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3190152 Dec  7 09:33 20221207_093105_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 3186772 Dec  7 09:39 20221207_093147_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2983828 Dec  7 09:39 20221207_093243_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2981123 Dec  7 09:45 20221207_093919_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2477438 Dec  7 10:12 20221207_100752_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2821559 Dec  7 10:12 20221207_100928_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2556828 Dec  7 10:15 20221207_101153_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2851558 Dec  7 10:18 20221207_101349_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2577042 Dec  7 10:18 20221207_101426_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2608394 Dec  7 10:42 20221207_103916_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2455134 Dec  7 10:42 20221207_104000_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 1064997 Dec  7 10:54 20221207_104913_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2269771 Dec  7 10:54 20221207_104950_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2514822 Dec  7 11:18 20221207_111510_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2792677 Dec  7 11:18 20221207_111547_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2584731 Dec  7 11:33 20221207_112800_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2881956 Dec  7 11:33 20221207_112839_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2873417 Dec  7 11:33 20221207_112915_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2885922 Dec  7 11:36 20221207_113124_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2895853 Dec  7 11:36 20221207_113207_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2679403 Dec  7 11:36 20221207_113249_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2897435 Dec  7 11:39 20221207_113418_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2941330 Dec  7 11:39 20221207_113454_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 1513292 Dec  7 11:39 20221207_113535_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2542815 Dec  7 12:18 20221207_121406_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 1981321 Dec  7 12:18 20221207_121525_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2788488 Dec  7 13:21 20221207_131908_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 1756391 Dec  7 13:39 20221207_133416_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2637336 Dec  7 13:39 20221207_133536_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2955239 Dec  7 13:51 20221207_134652_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2355881 Dec  7 14:00 20221207_135627_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2739410 Dec  7 14:03 20221207_135936_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  356813 Dec  7 14:03 20221207_135942_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2866643 Dec  7 14:03 20221207_140030_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2505464 Dec  7 16:06 20221207_155921_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 2010034 Dec  7 16:06 20221207_155958_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 1292905 Dec  7 17:24 20221207_171949_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 1158875 Dec  7 17:33 20221207_172804_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  332907 Dec  7 17:39 20221207_173655_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  173430 Dec  7 17:39 20221207_173711_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue 1329037 Dec  7 17:51 20221207_174958_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  615989 Dec  7 21:21 20221207_211846_Kitchen.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  906116 Dec  8 07:04 20221208_063706_SouthYard.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  442887 Dec  8 06:45 20221208_064052_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  415277 Dec  8 06:45 20221208_064143_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  283051 Dec  8 06:45 20221208_064158_BackDoor.mp4
-rw-rw-r-- 1 brent.perdue brent.perdue  512331 Dec  8 09:30 20221208_092537_Kitchen.mp4
```